crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -561,6 +562,58 @@ impl Mesh {
         Ok(self)
     }
 
+    /// Transforms the vertex positions, normals, and tangents of the mesh by the given [`Transform`].
+    pub fn transformed_by(mut self, transform: Transform) -> Self {
+        self.transform_by(transform);
+        self
+    }
+
+    /// Transforms the vertex positions, normals, and tangents of the mesh in place by the given [`Transform`].
+    pub fn transform_by(&mut self, transform: Transform) {
+        debug_assert!(
+            transform.scale.x == 0.0 && (transform.scale.y == 0.0 || transform.scale.z == 0.0)
+                || transform.scale.y == 0.0 && transform.scale.z == 0.0,
+            "mesh transform scale cannot be zero on more than one axis"
+        );
+
+        if let Some(VertexAttributeValues::Float32x3(ref mut positions)) =
+            self.attribute_mut(Mesh::ATTRIBUTE_POSITION)
+        {
+            // Apply scale, rotation, and translation to vertex positions
+            positions
+                .iter_mut()
+                .for_each(|pos| *pos = transform.transform_point(Vec3::from_slice(pos)).to_array());
+        }
+
+        // No need to transform normals or tangents if rotation is near identity and scale is uniform
+        if transform.rotation.is_near_identity()
+            && transform.scale.x == transform.scale.y
+            && transform.scale.y == transform.scale.z
+        {
+            return;
+        }
+
+        if let Some(VertexAttributeValues::Float32x3(ref mut normals)) =
+            self.attribute_mut(Mesh::ATTRIBUTE_NORMAL)
+        {
+            // Transform normals, taking into account non-uniform scaling and rotation
+            normals.iter_mut().for_each(|normal| {
+                let n = Vec3::from_slice(normal) * transform.scale.yzx() * transform.scale.zxy();
+                *normal = (transform.rotation * n.normalize()).to_array();
```

Comment:
use normalize_or_zero to handle cases when the normal is perpendicular to a zero-scale axis